testsuite/integration/manualmode/src/test/java/org/jboss/as/test/manualmode/messaging/mgmt/JMSBridgeManagementTestCase.java

```diff
@@ -0,0 +1,360 @@
+            result = setUpQueue(managementClient.getControllerClient(), "targetQueue", ADDRESS_QUEUE_TARGET);
+            Assert.assertTrue(result.toString(), Operations.isSuccessfulOutcome(result));
+        }
+
+        @Override
+        public void tearDown(final ManagementClient managementClient, final String containerId) throws Exception {
+            managementClient.getControllerClient().execute(Util.createRemoveOperation(ADDRESS_QUEUE_SOURCE));
+            managementClient.getControllerClient().execute(Util.createRemoveOperation(ADDRESS_QUEUE_TARGET));
+        }
+
+        private ModelNode setUpQueue(final ModelControllerClient client, final String name, final PathAddress address)
+                throws IOException {
+            final ModelNode operation = Util.createAddOperation(address);
+            operation.get("entries").add("java:jboss/jms/queue/" + name).add("java:jboss/exported/jms/queue/" + name);
+            operation.get("durable").set(true);
+            return client.execute(operation);
+
+        }
+    }
+
+    class DisableHornetQSecruity implements ServerSetupTask {
+        // read/store and in case 'true', change it back to it, since this test does not need security
+        private boolean oldValue = true;
+
+        @Override
+        public void setup(ManagementClient managementClient, String containerId) throws Exception {
+            final ModelControllerClient client = managementClient.getControllerClient();
+            ModelNode operation = Util.createOperation(READ_ATTRIBUTE_OPERATION, PathAddress.pathAddress(ADDRESS_HORNETQ));
+            operation.get(NAME).set(SECURITY_ENABLED.getName());
+            ModelNode result = client.execute(operation);
+            this.oldValue = result.get(RESULT).asBoolean();
+            if (this.oldValue) {
+                operation = Util.createOperation(WRITE_ATTRIBUTE_OPERATION, PathAddress.pathAddress(ADDRESS_HORNETQ));
+                operation.get(NAME).set(SECURITY_ENABLED.getName());
+                operation.get(VALUE).set(false);
+                result = client.execute(operation);
+                Assert.assertTrue("Failed to disable security in HornetQ", Operations.isSuccessfulOutcome(result));
+            }
+        }
+
+        @Override
+        public void tearDown(ManagementClient managementClient, String containerId) throws Exception {
+            if (this.oldValue) {
+                ModelNode operation = Util.createOperation(WRITE_ATTRIBUTE_OPERATION, PathAddress.pathAddress(ADDRESS_HORNETQ));
+                operation.get(NAME).set(SECURITY_ENABLED.getName());
+                operation.get(VALUE).set(true);
+                ModelNode result = managementClient.getControllerClient().execute(operation);
+                Assert.assertTrue("Failed to enable security in HornetQ", Operations.isSuccessfulOutcome(result));
+            }
+        }
+
+    }
+
+    class SetupJMSBridge implements ServerSetupTask {
```

Comment:
org.jboss.as.test.integration.common.jms.JMSOperations could be used to create the JMS bridge